File: live_price_frontend/lib/modules/parities/models/parity_create_model.dart
import 'package:live_price_frontend/modules/parities/models/spread_rule_type.dart';

class ParityCreateModel {
  final String name;
  final String symbol;
  final String rawSymbol;
  final bool isEnabled;
  final int orderIndex;
  final int scale;
  final SpreadRuleType spreadRuleType;
  final double spreadForAsk;
  final double spreadForBid;
  final int parityGroupId;

  ParityCreateModel({
    required this.name,
    required this.symbol,
    required this.rawSymbol,
    required this.isEnabled,
    required this.orderIndex,
    required this.scale,
    required this.spreadRuleType,
    required this.spreadForAsk,
    required this.spreadForBid,
    required this.parityGroupId,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'symbol': symbol,
      'rawSymbol': rawSymbol,
      'isEnabled': isEnabled,
      'orderIndex': orderIndex,
      'scale': scale,
      'spreadRuleType': spreadRuleType.id,
      'spreadForAsk': spreadForAsk,
      'spreadForBid': spreadForBid,
      'parityGroupId': parityGroupId,
    };
  }
}

File: LivePriceBackend/DTOs/Parity/ParityCreateModel.cs
using LivePriceBackend.Constants.Enums;

namespace LivePriceBackend.DTOs.Parity;

public class ParityCreateModel
{
    public required string Name { get; set; }
    public required string Symbol { get; set; }
    public required string RawSymbol { get; set; }
    public bool IsEnabled { get; set; }
    public int OrderIndex { get; set; }
    public int Scale { get; set; }
    public SpreadRuleType? SpreadRuleType { get; set; }
    public decimal? SpreadForAsk { get; set; }
    public decimal? SpreadForBid { get; set; }
    public int ParityGroupId { get; set; }
} 